tests/robustness/model/deterministic.go

```diff
@@ -93,8 +94,15 @@ func (s etcdState) step(request EtcdRequest) (etcdState, MaybeEtcdResponse) {
 	s.KeyValues = newKVs
 	switch request.Type {
 	case Range:
-		resp := s.getRange(request.Range.Key, request.Range.RangeOptions)
-		return s, MaybeEtcdResponse{EtcdResponse: EtcdResponse{Range: &resp, Revision: s.Revision}}
+		if request.Range.Revision == 0 || request.Range.Revision == s.Revision {
+			resp := s.getRange(request.Range.Key, request.Range.RangeOptions)
+			return s, MaybeEtcdResponse{EtcdResponse: EtcdResponse{Range: &resp, Revision: s.Revision}}
+		} else {
+			if request.Range.Revision > s.Revision {
+				return s, MaybeEtcdResponse{Err: EtcdFutureRevErr}
```

Comment:
Is this just a safety catch or are there some fully expected situations where this future rev is going to be hit?  Apologies if I am missing something obvious.

Reply:
It's a dummy error that should not be hit in the current code, still in future we will want to generate traffic that might hit a future revision and test it.